### ISSUE TITLE
Enhancement: Needs at least 2 free skill slots for role skills w/ 7 free sub-skill slots 

### ISSUE THREAD

**Issue by TekDragon** [ray-cw/roll20-CPR-charactersheet#3]
Every class gets a role-specific skill they can put points into. A multi-class character will have 2 role-specific skills. 

Some classes get sub-skills they put points into. Tech has the most, with 4, followed by Paramedic, with 3. 

Space for 2 role-specific skills (that do subtract from skill total) and up to 7 sub-skills (that don't subtract from skill total) would give players all the room they need to work with.

**Comment by ray-cw** [comment]
Good feature, though really, can't a character multiclass as many times as they want RAW? I should be making it so you can add any number of role-specific skills

**Comment by TekDragon** [comment]
Yes, you're right. Perhaps the easiest way to do it is just provide players an area at the bottom with a bunch of class skill fields and players can maintain it. 

If you go that route, you'll want to maintain consistency by bolding the very first field and adding 4 points to it, since a player has to start with 4 skill points in their starting class skill. No fields after the first need to be factored into the skill pool. That way techs and paramedics can use the remaining fields for keeping track of their sub-skills without it screwing with the pool.

**Comment by ray-cw** [comment]
![image](https://user-images.githubusercontent.com/28985202/100976129-164e2880-350d-11eb-952a-1e6e9ea1272c.png)

This is currently what I have going so far:

- A tab for a player to maintain multiple roles
- For each role, a player gets 6 Subskills and 6 rolls that they can define themselves.
- For Rolls, I'm planning on having a dropdown for stats to choose from, Ranks the player can fill in themself.... Skill, implementing a number input would be easiest for me, as implementing a dropdown for ALL stats seems daunting

Addressing on your point about automatically starting with 4 skill points in their starting class skill; I'm not sure the best way to go about that in the Javascript code and quite frankly this character sheet can't hand hold players all the way.

**Comment by TekDragon** [comment]
Wow, that looks fantastic! Good call on the 6 sub-skills, I forgot about the Solo. 

And I agree, they can figure it out.
